src/AzureAuth/Commands/Ado/CommandPat.cs

```diff
@@ -60,19 +97,149 @@ private enum OutputMode
         [Option(ScopeOption, ScopeHelp, CommandOptionType.MultipleValue)]
         private string[] Scopes { get; set; } = null;
 
+        [Option(PromptHintOption, PromptHintHelp, CommandOptionType.SingleValue)]
+        private string PromptHint { get; set; } = null;
+
         [Option(OutputOption, OutputHelp, CommandOptionType.SingleValue)]
         private OutputMode Output { get; set; } = OutputMode.Token;
 
+        [Option(DomainOption, DomainHelp, CommandOptionType.SingleValue)]
+        private string Domain { get; set; } = AzureAuth.Ado.Constants.PreferredDomain;
+
         /// <summary>
         /// Executes the command and returns a status code indicating the success or failure of the execution.
         /// </summary>
         /// <param name="logger">The <see cref="ILogger{T}"/> instance that is used for logging.</param>
+        /// <param name="publicClientAuth">An <see cref="IPublicClientAuth"/>.</param>
         /// <param name="eventData">Lasso injected command event data.</param>
         /// <returns>An integer status code. 0 for success and non-zero for failure.</returns>
-        public int OnExecute(ILogger<CommandPat> logger, CommandExecuteEventData eventData)
+        public int OnExecute(ILogger<CommandPat> logger, IPublicClientAuth publicClientAuth, CommandExecuteEventData eventData)
         {
-            logger.LogInformation("coming soon");
+            if (!this.ValidOptions(logger))
+            {
+                return 1;
+            }
+
+            var accessToken = this.AccessToken(publicClientAuth, eventData);
+            if (accessToken == null)
+            {
+                logger.LogError("Failed to acquire an Azure DevOps access token. Re-run with '--verbosity debug' for more info.");
+                return 1;
+            }
+
+            var cache = this.Cache();
+            var client = this.Client(accessToken.Token);
+            var manager = new PatManager(cache, client);
+
+            using (new CrossPlatLock(LockfilePath))
+            {
+                var pat = manager.GetPatAsync(this.PatOptions()).Result;

```

Comment:
Reason we don't write as `await manager.GetPatAsync(this.PatOptions);`

await asynchronously unwraps the result of your task, whereas just using Result would block until the task had completed.

Reply:
If we addressed this I think it would be good to do as a follow up PR. We could make `PatManager.GetPatAsync` actually just be synchronous and then there wouldn't be a need to use `.Result` at this point.

Reply:
For now, this is working, so I'm inclined to leave it as is.